eigentrust-zk/src/circuits/opinion/native.rs

```diff
@@ -72,8 +72,11 @@ where
 		let mut hashes = Vec::new();
 		for i in 0..NUM_NEIGHBOURS {
 			let att = self.attestations[i].clone();
-			assert!(att.attestation.about == set[i]);
-			assert!(att.attestation.domain == self.domain);
+
+			if att != SignedAttestation::empty(self.domain) {
+				assert!(att.attestation.about == set[i]);
+				assert!(att.attestation.domain == self.domain);
+			}
```

Comment:
We can't have this since we want the logic to match the circuit, and the logic should be that all attestations should pass this check:
```rust
assert!(att.attestation.about == set[i]);
assert!(att.attestation.domain == self.domain);
```

The scores are then nullified in case the signature verification fails.

So, we should do the same here. When we are constructing the `Attestation::empty`, the about address should be passed as well (should be taken from the set). This logic should be done here:
https://github.com/eigen-trust/protocol/blob/fix/scores-calc/eigentrust-zk/src/circuits/dynamic_sets/native.rs#L198-L199